### PR TITLE
Create aspnetcore shared framework symbols packages

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/aspnet/Universe/issues/853

Checking effects on downstream builds.

### CODE REVIEW

File: build/SharedFx.targets [hunk 1]
```diff
@@ -383,4 +422,45 @@
   </Target>
 
   <Target Name="BuildSharedFx" DependsOnTargets="ResolveSharedFxFiles;CrossGenAssemblies;CrossGenSymbols;PackSharedFx"/>
+
+  <Target Name="_BuildSharedFxSymbols">
+    <!-- Clear working directory -->
+    <RemoveDir Directories="$(_WorkRoot)" />
+
+    <ItemGroup>
+      <RuntimeSymbolsArchive Include="win-x86;win-x64;osx-x64;linux-x64"/>
```
Comment (natemcmaster) [inline build/SharedFx.targets]: This is already defined in repo.props.

File: build/SharedFx.targets [hunk 2]
```diff
@@ -383,4 +422,45 @@
...
+    </ItemGroup>
+
+    <!-- Extract Symbols archives -->
+    <UnzipArchive
+      File="$(_SymbolsSourceDir)$(SymbolsSourceArchiveName)-%(RuntimeSymbolsArchive.Identity).zip"
```
Comment (natemcmaster) [inline build/SharedFx.targets]: Can you clarify what's going on here? Why are we producing a zip of symbols, unzipping it, and then packing it into a nuspec?

File: build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <id>Microsoft.AspNetCore.All.symbols</id>
+    <version>$VERSION$</version>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <licenseUrl>http://www.microsoft.com/web/webpi/eula/net_library_eula_ENU.htm</licenseUrl>
+    <projectUrl>http://www.asp.net/</projectUrl>
+    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
+    <description>Symbol packages for Microsoft.AspNetCore.All shared framework</description>
+    <copyright>© .NET Foundation. All rights reserved.</copyright>
```
Comment (natemcmaster) [inline build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec]: Can you update copyright and licenseUrl to match our runtime packages? 

https://github.com/aspnet/BuildTools/blob/dev/src/Internal.AspNetCore.Sdk/build/Common.props#L13-L22

I'd actually prefer to flow these automatically from Internal.AspNetCore.Sdk so we don't have to remember to update more places.

File: build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    <tags>aspnetcore</tags>
+    <serviceable>true</serviceable>
+    <repository type="git" url="https://github.com/aspnet/Universe" />
+  </metadata>
+  <files>
+    <file src="*.pdb" target="" />
```
Comment (natemcmaster) [inline build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec]: Should these be nested according to RID? Are there symbols for multiple RIDs in the package?

File: build/SharedFx.targets (hunk 2 above)
Comment (JunTaoLuo) [reply build/SharedFx.targets]: The symbols are built on each platform win-x64/win-x86/linux-x64/osx-x64. Then we need another configuration that takes the symbols produced on each platform and pack them into a symbols nupkg for .App and .All.

Comment (natemcmaster) [reply build/SharedFx.targets]: Maybe we should rid split our symbols packages... I'm thinking ahead to our goal to build aspnetcore on a single machine. If we can avoid introducing more build outputs that require building on 3 machines and still satisfy the symbols package requirement, I think we should do that.

File: build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec (hunk 3 above)
Comment (JunTaoLuo) [reply build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec]: How do we automatically flow these from Internal.AspNetCore.Sdk? Add them all to properties passed into PackNuspec?

Comment (natemcmaster) [reply build/tools/templates/SharedFxSymbols/Microsoft.AspNetCore.All.symbols.nuspec]: Build the symbols nupkg from a MSBuild project that references Internal.AspNetCore.Sdk.

**Comment by JunTaoLuo** [comment]
:up::date:

**Pull request merged**